honeybee_radiance/lightsource/sky/sunmatrix.py

```diff
@@ -0,0 +1,192 @@
+"""Generate a point-in-time climate-based sky."""
+from __future__ import division
+import honeybee.typing as typing
+import ladybug.futil as futil
+from ladybug.wea import Wea
+import os
+
+
+class SunMatrix(object):
+    """Annual Climate-based Sun matrix.
+
+    The output of SkyMatrix is similar to using command Radiance's gendaymtx command with
+    ``-n -D`` options. The options are available in Radiance 5.3 and after that. For more
+    information see gendaymtx documentation.
+
+    https://www.radiance-online.org/learning/documentation/manual-pages/pdfs/gendaymtx.pdf
+
+    Args:
+        wea: A Ladybug wea object.
+        north: A number between -360 and 360 for the counterclockwise difference between
+            the North and the positive Y-axis in degrees. 90 is West and 270 is East
+            (Default: 0)
+
+    Properties:
+        * wea
+        * location
+        * north
+        * is_point_in_time
+        * is_climate_based
+    """
+
+    __slots__ = ('_wea', '_north')
+
+    def __init__(self, wea, north=0):
+        """Create a climate-based sun matrix."""
+        self.wea = wea
+        self.north = north
+
+    @property
+    def wea(self):
+        """Get and set wea."""
+        return self._wea
+
+    @wea.setter
+    def wea(self, value):
+        assert isinstance(value, Wea), \
+            'wea must be from type Wea not {}'.format(type(value))
+        self._wea = value
+
+    @property
+    def north(self):
+        """Get and set north direction.
+
+        A number between -360 and 360 for the counterclockwise difference between
+        the North and the positive Y-axis in degrees. 90 is West and 270 is East.
+        """
+        return self._north
+
+    @north.setter
+    def north(self, value):
+        value = typing.float_in_range(value, -360, +360, 'Skymatrix north')
+        self._north = value
```

Comment:
This is just a recommendation but, if you want to expose a method to set the north using a vector, it's very easy to do with ladybug_geometry.  You can see some sample code here on the honeybee-core Model:
https://github.com/ladybug-tools/honeybee-core/blob/master/honeybee/model.py#L238-L239

Reply:
For now I keep it as a number and leave the vector based input conversion to numbers to each interface.